pages/summary.js

```diff
@@ -2,28 +2,78 @@ import PropTypes from "prop-types";
 import { API_URL } from "../lib/constants";
 import { getQueryStringFromServerQuery } from "../lib/query-utils";
 import Breadcrumbs from "../components/breadcrumbs";
-import JsonDisplay from "../components/json-display";
+import { DataPanel, DataAreaTitle } from "../components/data-area";
+import { ButtonLink } from "../components/form-elements";
 import PagePreamble from "../components/page-preamble";
+import RegulomeVersionTag from "../components/regulome-version-tag";
+import SummaryTable from "../components/summary-table";
+import Notifications from "../components/notifications";
 
-export default function Summary({ data }) {
+export default function Summary({ data, queryString }) {
+  const total = data.total ? data.total : 0;
+  const variants = data.variants;
+  const assembly = data.assembly;
+  if (total > 1) {
+    for (let i = 0; i < variants.length; i++) {
+      variants[
+        i
+      ].chrom_location = `${variants[i].chrom}:${variants[i].start}-${variants[i].end}`;
+      variants[i].rank = variants[i].regulome_score.ranking;
+      variants[i].score = variants[i].regulome_score.probability;
+    }
+  }
```

Comment:
This mutates the `data` prop which [React discourages](https://react.dev/learn/keeping-components-pure#recap). The method to fix this depends on the data’s complexity and personal preference. The obvious way is copying `data.variants` and mutating the copy. If `data.variants` is somewhat complicated, this could be a pain. Copying with `const variantsCopy = {...data.variants}` does a shallow copy, so this works if `data.variants` has no embedded objects and arrays, but not if it does unless you don’t mutate those embedded objects and arrays. Another possibility is generating a parallel array to `data.variants` that holds the location, rank, and score.

Reply:
I won't change the embedded objects and arrays so I will make a shallow copy. Thank you!

Reply:
mmmmm. shallow copy of array here is not working for me. I will generate a new array then.